lib/Monads/NonDetMonad.thy

```diff
@@ -606,6 +606,9 @@ notation (output)
 
 section "Combinators that have conditions with side effects"
 
+definition notM :: "('s, bool) nondet_monad \<Rightarrow> ('s, bool) nondet_monad" where
+  "notM m = do c \<leftarrow> m; return (\<not> c) od"
```

Comment:
Not sure if this is easier or more obscure, but since it's a unary function, we could just define `notM = liftM Not`